src/server/internalServer.cpp

```diff
@@ -462,18 +462,19 @@ bool InternalServer::start() {
   sockAddr6.sin6_port = htons(m_port);
 
   if (m_addr.empty()) {
-    if (0 != INADDR_ANY) {
+    if (0 != INADDR_ANY) { // what is the purpose of this? INADDR_ANY is always 0. Refer: http://www.castaglia.org/proftpd/doc/devel-guide/src/include/inet.h.html
       sockAddr6.sin6_addr = in6addr_any;
       sockAddr4.sin_addr.s_addr = htonl(INADDR_ANY);
     }
-    m_addr = kiwix::getBestPublicIp(m_ipMode == IpMode::ipv6 || m_ipMode == IpMode::all);
+    m_addr = kiwix::getBestPublicIps(m_ipMode);
   } else {
-    bool ipv6 = inet_pton(AF_INET6, m_addr.c_str(), &(sockAddr6.sin6_addr.s6_addr)) == 1;
-    bool ipv4 = inet_pton(AF_INET, m_addr.c_str(), &(sockAddr4.sin_addr.s_addr)) == 1;
+    // TODO: Needs more work
```

Comment:
What else needs to be done?

Reply:
This was a comment left to check if everything is ok after the change, before the entire scope was finalized. It is ok like this. 

Will remove comment.